### PR TITLE
Create README, add entry points and document "operator client"

### PR DESCRIPTION
**Pull request by OlivierChirouze**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,23 +23,7 @@ It looks like this:
...
-| [operator-design.md](./mvp-spec/operator-design.md)                                   | Design of the generation of Prebid SSO Data.                              |
-| [operator-design-alternative-swan.md](./mvp-spec/operator-design-alternative-swan.md) | Summary of the SWAN solution for generating PAF Data.                     |
-| [operator-requirements.md](./mvp-spec/operator-requirements.md)                       | Requirements for the generation of the PAF Data.                          |
-| [website-design.md](./mvp-spec/website-design.md)                                     | Design of the website integration.                                        |
-| [model](./mvp-spec/model)                                                             | Data and messages model                                                                             |
-| [json-schemas](./mvp-spec/json-schemas)                                               | Data and messages model in [JSON schema](https://json-schema.org/understanding-json-schema/) format |
+(e.g. operator1.onekey.network and operator2.onekey.network).
```
Comment (OlivierChirouze) [inline README.md:26]: Notice updated domains

File: mvp-spec/advertiser-implementation.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Advertiser
```
Comment (OlivierChirouze) [inline mvp-spec/advertiser-implementation.md:1]: Of course, we will need to add more content to this document in future PRs

File: mvp-spec/cmp-implementation.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# CMP (Consent Management Platform)
```
Comment (OlivierChirouze) [inline mvp-spec/cmp-implementation.md:1]: Of course, we will need to add more content to this document in future PRs

File: mvp-spec/publisher-implementation.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Publisher
```
Comment (OlivierChirouze) [inline mvp-spec/publisher-implementation.md:1]: Of course, we will need to add more content to this document in future PRs

File: mvp-spec/operator-client.md [hunk 5]
```diff
@@ -0,0 +1,576 @@
...
+{
+  "message": "Invalid signature"
+}
+```
+<!--partial-end-->
+
+</details>
+
+## Frontend operator client
+
```
Comment (OlivierChirouze) [inline mvp-spec/operator-client.md:413]: Here, I plan to add a diagram that explains when the frontend operator client calls the backend operator proxy.
(for another PR)

**Comment by lbdvt** [comment]
Thanks for this PR! Clear documentation is always very useful.

I would not do a page per participant type, and avoid duplicating content through inserts, as it makes it more difficult for the reader to understand what she may have already read or not.

I would suggest having the following pages:
- Workflows: which would explicit which endpoints each participant is expected to use or expose
- Operator: endpoints exposed by the operator
- Participants: endpoints exposed by the participants
- Transmissions signature requirements
- Audit log requirements

**Comment by lbdvt** [comment]
Thanks for this PR! Clear documentation is always very useful.

I would not do a page per participant type, and avoid duplicating content through inserts, as it makes it more difficult for the reader to understand what she may have already read or not.

I would suggest having the following pages:
- Workflows: which would explicit which endpoints each participant is expected to use or expose
- Operator: endpoints exposed by the operator
- Participants: endpoints exposed by the participants
- Transmissions signature requirements
- Audit log requirements

**Comment by OlivierChirouze** [comment]
Thanks for your comment @lbdvt .
I get your point.
I will re-organise to remove the individual documents I created, but I won't handle the next steps you mention in this PR.
Let's re-discuss this for another PR.

**Comment by OlivierChirouze** [comment]
@lbdvt please confirm if you agree with the latest changes and strategy to handle next steps in another PR.
If so, please approve the PR (if you have permissions. Otherwise we will make sure you get it)
Thanks

**Review by RomainLofaso — COMMENTED** [review]
Why so many files are impacted?

**Review by lbdvt — APPROVED** [review]

**Comment by OlivierChirouze** [comment]
@RomainLofaso see https://github.com/criteo/paf-mvp-implementation/issues/35

File: mvp-spec/README.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+## CMP (Consent Management Platform)
+
+Usually, publishers include a CMP javascript in their web pages, to gather user consent _on their behalf_.
+
+In PAF, the CMP is also responsible for **signing user preferences** and **writing** these preferences via the operator.
+
+Note that the CMP role can be taken by the publisher in some cases.
+
+See [cmp-implementation.md](cmp-implementation.md).
+
```
[the remark this answers is not in the document]
Comment (OlivierChirouze) [reply mvp-spec/README.md:72]: I agree this text is confusing. I will use the "User Interface Provider" terminology instead.

File: mvp-spec/README.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+1. Just like the advertiser, it needs to read id and preferences from the operator
+2. It is also selling inventory to contracting parties and must create and sign a "seed" object and initialize an RTB transaction sent to an SSP
+   1. this is done through the ad server
+
+See [publisher-implementation.md](publisher-implementation.md).
+
+## Ad server
+
+See [ad-server-implementation.md](ad-server-implementation.md).
+
```
[the remark this answers is not in the document]
Comment (OlivierChirouze) [reply mvp-spec/README.md:88]: I will rephrase to refer to "nodes" instead and adjust the wordings where needed.
For the moment I would like to keep further "refactoring" of the documents for a separate PR.

**Comment by OlivierChirouze** [comment]
@lbdvt rewording done

**Comment by OlivierChirouze** [comment]
@RomainLofaso I've managed to update the script that generates examples so that it only replace a file if the message is different (ignoring the signature).
This way, when we regenerate examples we will see only changes for messages which body has changed.

See https://github.com/criteo/paf-mvp-implementation/pull/41

In this PR I had to "reset" all files but from now on we should not be "polluted" anymore!

**Review by lbdvt — APPROVED** [review]

**Pull request merged**